Automattic/MobilePay-Apple
# Map custom errors

## Description

This PR maps custom API errors. This enables us to return more meaningful error messages to whoever is consuming MobilePayKit.

Previously, all API errors were being propagated as decoding errors, since we weren't mapping custom errors.

Before | After
--- | ---
`Error fetching products: The data couldn’t be read because it isn’t in the correct format.` | `Error fetching products: The OAuth2 token is invalid.`

## Notes

I also added OHHTTPStubs to MobilePayKit's test target to make it easier to  stub / write tests.

## How to test

1. Run either the iOS target or the macOS target without configuring the token
2. ✅ Notice the error message in the console reads: `Error fetching products: The OAuth2 token is invalid.`

Also, make sure tests are green.
